docs/logging-usage.rst

```diff
@@ -392,70 +392,105 @@ Delete a sink:
 
-    >>> handler = CloudLoggingHandler(client, name="mycustomlog")
+   >>> client.setup_logging(log_level=logging.INFO, excluded_loggers=('werkzeug',)))
 
-It is also possible to attach the handler to the root Python logger, so that for example a plain
-`logging.warn` call would be sent to Cloud Logging, as well as any other loggers created. However,
-you must avoid infinite recursion from the logging calls the client itself makes. A helper
-method :meth:`setup_logging <google.cloud.logging.handlers.setup_logging>` is provided to configure
-this automatically:
+CloudLoggingHandler
+=======================
+
+If you prefer not to use :meth:`google.cloud.logging.client.get_default_handler`, you can directly
+create a :class:`~CloudLoggingHandler <google.cloud.logging.CloudLoggingHandler>` instance which
+will write directly to the API.
 
 .. code-block:: python
 
     >>> import logging
     >>> import google.cloud.logging # Don't conflict with standard logging
-    >>> from google.cloud.logging.handlers import CloudLoggingHandler, setup_logging
+    >>> from google.cloud.logging.handlers import CloudLoggingHandler
     >>> client = google.cloud.logging.Client()
     >>> handler = CloudLoggingHandler(client)
-    >>> logging.getLogger().setLevel(logging.INFO) # defaults to WARN
-    >>> setup_logging(handler)
-    >>> logging.error('bad news')
+    >>> cloud_logger = logging.getLogger('cloudLogger')
+    >>> cloud_logger.setLevel(logging.INFO) # defaults to WARN
+    >>> cloud_logger.addHandler(handler)
+    >>> cloud_logger.error('bad news')
 
-You can also exclude certain loggers:
+.. note::
+
+    This handler by default uses an asynchronous transport that sends log entries on a background
+    thread. However, the API call will still be made in the same process. For other transport
+    options, see the transports section.
+
+All logs will go to a single custom log, which defaults to "python". The name of the Python
+logger will be included in the structured log entry under the "python_logger" field. You can
+change it by providing a name to the handler:
 
 .. code-block:: python
 
-   >>> setup_logging(handler, excluded_loggers=('werkzeug',)))
+    >>> handler = CloudLoggingHandler(client, name="mycustomlog")
+
+fluentd logging handlers
+==============================
+
+Besides :class:`~google.cloud.logging.handlers.CloudLoggingHandler`, which writes directly to
+the API, two other handlers are provided. :class:`~google.cloud.handlers.GAEHandler.GAEHandler`,
+ which is recommended when running on the Google App Engine Flexible vanilla runtimes, and
+:class:`~google.cloud.handlers.GKE.GKEHandler`, which is recommended when running on Google
+Container Engine with the Stackdriver Logging plugin enabled.
 
+:meth:`~google.cloud.logging.client.get_default_handler` and
+:meth:`~google.cloud.logging.client.setup_logging` will attempt to use
+environment variables to automatically detect whether the code is running in these platforms and use
```

Comment:
Just environment variables? Won't it use other aspects of the environment, e.g. metadata server or the presence of importable libraries?